native/src/main/java/io/ballerina/lib/data/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,271 @@
+        } else {
+            BArray array = toJsonArray(obj);
+            long index;
+            if (key != null) {
+                index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+            } else {
+                index = array.size();
+            }
+
+            if (index == array.size()) {
+                array.append(createJsonElement(value));
+            } else {
+                array.add(index, createJsonElement(value));
+            }
+        }
+    }
+
+    public void removeProperty(Object obj, Object key) {
+        if (isMap(obj)) {
+            toJsonObject(obj).remove(key.toString());
+        } else {
+            BArray array = toJsonArray(obj);
+            int index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+            array.shift(index);
+        }
+    }
+
+    public boolean isMap(Object obj) {
+        return obj instanceof BMap;
+    }
+
+    public Collection<String> getPropertyKeys(Object obj) {
+        return Arrays.stream(toJsonObject(obj).getKeys())
+                .map(Object::toString)
+                .collect(Collectors.toList());
+    }
+
+    public int length(Object obj) {
+        if (isArray(obj)) {
+            return toJsonArray(obj).size();
+        } else if (isMap(obj)) {
+            return toJsonObject(obj).entrySet().size();
+        } else if (isJsonPrimitive(obj)) {
+            return obj.toString().length();
+        }
+        throw new JsonPathException(
+                "length operation cannot be applied to " + (obj != null ? obj.getClass().getName() : "null"));
+    }
+
+    public Iterable<?> toIterable(Object obj) {
+        BArray arr = toJsonArray(obj);
+        List<Object> values = new ArrayList<>(arr.size());
+        for (int i = 0; i < arr.size(); i++) {
+            values.add(unwrap(arr.get(i)));
+        }
+        return values;
+    }
+
+    public Object unwrap(Object obj) {
+        if (obj == null) {
+            return null;
+        }
+        Type targetType;
+        if (isMap(obj)) {
+            targetType = PredefinedTypes.TYPE_JSON;
+        } else if (isArray(obj)) {
+            targetType = PredefinedTypes.TYPE_JSON_ARRAY;
+        } else if (obj instanceof BString) {
+            targetType = PredefinedTypes.TYPE_STRING;
+        } else if (obj instanceof BDecimal) {
+            targetType = PredefinedTypes.TYPE_FLOAT;
+        } else if (obj instanceof String) {
+            targetType = PredefinedTypes.TYPE_STRING;
+        } else if (obj instanceof Boolean) {
+            targetType = PredefinedTypes.TYPE_BOOLEAN;
+        } else if (obj instanceof Long) {
+            targetType = PredefinedTypes.TYPE_INT;
+        } else if (obj instanceof Integer) {
+            obj = ((Integer) obj).longValue();
+            targetType = PredefinedTypes.TYPE_INT;
+        } else if (obj instanceof BigDecimal) {
+            targetType = PredefinedTypes.TYPE_JSON_DECIMAL;
```

Comment:
Shall we add a test for this

Reply:
I added tests for Decimals and this line not get covered, I will check it and remove or add a test in a separate PR.

Reply:
https://github.com/ballerina-platform/module-ballerina-data.jsondata/issues/5 to track this

Reply:
Library issues are tracked in https://github.com/ballerina-platform/ballerina-library/issues, Could you pls move this issue there.